doc/lsst.meas.base/intro.rst

```diff
@@ -0,0 +1,94 @@
+
+meas_base provides those two drivers and a bunch of simple algorithms.
+Other packages can provide more of either, following the same conventions.
+Most of our meas_extensions packages provide measurement algorithms for use in the meas_base framework (e.g. simpleShape, shapeHSM, etc).
+
+A notable extension package is ip_diffim.
+This provides a driver suitable for meas_base-style measurement on difference images and a bunch of algorithms (e.g. dipole measurement) which are of relevance in that context.
+
+Naming and selecting measurement algorithms
+===========================================
+
+The new users can generally skip over the details of measurement drivers: we have the ones we need for now, and they mostly “just work”.
+The interesting stuff all happens in algorithms.
+
+Understand first that algorithms may be referred to by strings which describe both their home package and their name.
+Thus, ``base_PsfFlux`` refers to the ``PsfFlux`` algorithm in :py:mod:`lsst.meas.base`; ``ip_diffim_DipoleFit`` refers to the ``DipoleFit`` algorithm in :py:mod:`lsst.ip.diffim`.
+
+When configuring a measurement driver (i.e. a task like :py:class:`lsst.meas.base.sfm.SingleFrameMeasurementTask`) we specify a list of plugins to be run when executing the driver.
+See, for example, `here`_.
+These can be over-ridden by the user in the task configuration.
+
+When the driver runs, it will return an ``afw::table``.
+Each row in the table contains the results of running all of the plugins which have been enabled on a particular source.
+Each algorithm plugin can add columns to the tables to contain its results.
+These columns build upon the naming scheme described above.
+Thus, if the ``base_PsfFlux`` algorithm is run, you will see a bunch of fields in the table with names starting ``base_PsfFlux_`` — ``base_PsfFlux_instFlux``, ``base_PsfFlux_instFluxErr``, and so on.
+
+(Incidentally, note that this means that the schema of the output table depends on the particular plugins that were enabled at task configuration time. This makes it awkward to provide a single document that describes pipeline outputs — they vary with user configuration.)
+
+Algorithm fields and flags
+==========================
+
+Following from the above, when a new measurement driver (task) is created, it will:
+
+#. Create a new schema for its output;
+#. Hand that to each of the measurement algorithms (plugins) it's planning to
+   run, and ask them to add fields which will be used to store their outputs.
+
+In the limit of poor documentation, we can look at the code for the algorithm to understand what fields it's adding and what they mean.
+(Alternatively, we can inspect the schema of the output table.)
+
+Note that algorithms can be implemented in C++ or Python, so we need to be able to look at both.
+
+`Here is`_ an example in Python: the ``base_FPPosition`` plugin.
```

Comment:
I'd put a subhead above this: "Inspecting the schema generation code of Python plugins"

Reply:
I'd make `base_FPPosition` itself the link text.